Chummer/frmCharacterOptions.cs

```diff
@@ -765,11 +822,43 @@ private void PopulateCustomDataDirectoryTreeView()
                 }
             }
 
-            if(objOldSelected != null)
+            //CheckForDependenciesAndExclusivities();
+
+            if (objOldSelected != null)
                 treCustomDataDirectories.SelectedNode = treCustomDataDirectories.FindNodeByTag(objOldSelected);
+            treCustomDataDirectories.ShowNodeToolTips = true;
             treCustomDataDirectories.EndUpdate();
         }
 
+        private void CheckForDependenciesAndExclusivities()
+        {
+            foreach (TreeNode directoryNode in treCustomDataDirectories.Nodes)
```

Comment:
The real check for dependencies and exclusivities is only done for nodes which are `checked`.

I did some basic diagnostics a while back and AFAICT it isn't really noticeable in regards to performance, even with a bunch of active directories.

A version in which it is only checked for recently changed once has the problem, that it doesn't update already active directories.
As an example: If I activate Directory "1", that is dependent on "2" it would show up as red.
Now I activate "2", but "1" would still be shown as missing something.

The object of `CustomDataDirectoryInfo` "2" doesn't know if some other directory may be dependent upon it, so we need to check every active dataset.

Reply:
I was thinking more in terms of setting up some kind of graph when all the `CustomDataDirectoryInfo`s are first created that links up all the different relations they have. Then when you change the enabled status or load order of one info, you just need to query all the infos connected to it in that graph. However, I admit that's probably a bit much for now and unnecessary to get the first working versions of these manifests usable by the public, so you can probably file that away as a TODO somewhere near this code and move on.